tests/client/amo/components/TestAddonReview.js

```diff
@@ -0,0 +1,197 @@
+    updateReviewText: () => {},
+    ...customProps,
+  };
+  const root = findRenderedComponentWithType(renderIntoDocument(
+    <I18nProvider i18n={props.i18n}>
+      <RouterStub router={fakeRouter}>
+        <AddonReviewBase {...props} />
+      </RouterStub>
+    </I18nProvider>
+  ), AddonReviewBase);
+
+  return findDOMNode(root);
+}
+
+describe('AddonReview', () => {
+  it('can update a review', () => {
+    const fakeRouter = {};
+    const updateReviewText = sinon.spy(() => {});
+    const rootNode = render({ updateReviewText, fakeRouter });
+
+    const textarea = rootNode.querySelector('textarea');
+    textarea.value = 'some review';
+    Simulate.submit(rootNode.querySelector('form'));
+
+    assert.equal(updateReviewText.called, true);
+    const params = updateReviewText.firstCall.args[0];
+    assert.equal(params.body, 'some review');
+    assert.equal(params.addonId, defaultReview.addonSlug);
+    assert.equal(params.reviewId, defaultReview.id);
+    assert.equal(params.apiState, signedInApiState);
+    assert.equal(params.router, fakeRouter);
+  });
+
+  it('requires the review text to be non-empty', () => {
+    const rootNode = render();
+    // By default the textarea for the review is empty.
+    try {
+      Simulate.submit(rootNode.querySelector('form'));
+      assert(false, 'unexpected success');
+    } catch (error) {
+      assert.match(error.message, /review .* cannot be empty/);
+    }
+  });
+
+  it('requires a review object', () => {
+    const review = { nope: 'not even close' };
+    try {
+      render({ review });
+      assert(false, 'unexpected success');
+    } catch (error) {
+      assert.match(error.message, /Unexpected review property: {"nope".*/);
+    }
+  });
+
+  describe('loadAddonReview', () => {
+    let mockApi;
+    let fakeDispatch;
+
+    beforeEach(() => {
+      mockApi = sinon.mock(coreApi);
+      fakeDispatch = sinon.spy(() => {});
+    });
+
+    function loadAddonReview({ params } = {}) {
+      let localParams = params;
+      if (!localParams) {
+        localParams = { slug: fakeAddon.slug, reviewId: defaultReview.id };
+      }
+      return defaultAddonReviewLoader(
+        { store: { dispatch: fakeDispatch }, params: localParams });
+    }
+
+    it('requires URL params',
+      () => loadAddonReview({ params: {} })
+        .then(() => assert(false, 'unexpected success'))
+        .catch((error) => {
+          assert.match(error.message, /missing URL params/);
+        })
+    );
+
+    it('loads a review', () => {
+      const reviewResponse = {
+        id: 7776,
+        addon: { id: 1234 },
+        version: { id: 4321 },
+        rating: 2,
+        user: { id: 9876 },
+      };
+
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          endpoint: `addons/addon/${fakeAddon.slug}/reviews/${defaultReview.id}`,
+          method: 'GET',
+        })
+        .returns(Promise.resolve(reviewResponse));
+
+      return loadAddonReview()
```

Comment:
I can't tell but it might be more clear coming back to this later if this function was called `defaultLoadAddonReview()` and it called `loadAddonReview()` instead of the other way around.